server/src/main/java/org/elasticsearch/client/internal/node/NodeClient.java

```diff
@@ -67,6 +69,13 @@ public void initialize(
         this.namedWriteableRegistry = namedWriteableRegistry;
     }
 
+    /**
+     * Return the names of all available actions registered with this client.
+     */
+    public List<String> getActionNames() {
+        return actions.keySet().stream().map(ActionType::name).collect(Collectors.toList());
+    }
```

Comment:
Trivially, `.collect(Collectors.toList())` could be replaced with `.toList()`, which has the advantage of creating an unmodifiable list. (additionally, if this suggestions is adopted, then the javadoc could reflect the unmodifiability of the returned list)